Add cookie logger tests

licitdev opened the pull request:
## Description

<!--

A summary of the changes and a reference to the Issue that was fixed / implemented.

NOTE: All Pull Requests require a corresponding open Issue.

Please reference the Issue number below:

-->

Add tests for #17914. Important to set `LOGGER_LEVEL` explicitly within tests that accesses the logger.

## Todo
- [x] Get `server.stdout.on('data'...)` working

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [x] Other, please describe: Tests

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

licitdev committed:
Update gql test transport to pass cookies

licitdev committed:
Add token refresh tests

licitdev committed:
Add logger redaction tests

licitdev committed:
Update gql expires type to string

licitdev committed:
Add redact test to sequential flow

licitdev committed:
Try increase delay to 250

licitdev committed:
Try increase delay to 500

licitdev committed:
Use passthrough

paescuj committed:
Await specific log lines

paescuj committed:
Merge branch 'main' into cookie-logger-tests

paescuj committed:
Cleanup listener in any case & proper log level

licitdev committed:
Extract logger for reusability

licitdev committed:
Merge branch 'main' into cookie-logger-tests

licitdev committed:
Rename file and use off instead

licitdev committed:
Add root upload folder to gitignore

Co-authored-by: Pascal Jufer <pascal-jufer@bluewin.ch>

Nitwel reviewed (approved):
Looks good to me!

br41nslug committed:
Merge branch 'main' into cookie-logger-tests

paescuj reviewed (changes requested):
Going to push some additions in a few minutes, please don't merge yet 🙏

paescuj committed:
Discard previous logs & add filter functionality

To be more sure that we only check the log lines we are really interested in, the previous data is now discarded (intercept logs from start of logger) and also there is now a filter function that makes that only logs with a certain condition are intercepted

The stop and filter conditions can now also be a function so that we can also easily use the class for future cases

paescuj reviewed (approved):
So far all log entries of the server were captured and checked, which could theoretically lead to false positives, since the conditions in the `expect`s could also be met by other requests,
Therefore the changes in https://github.com/directus/directus/pull/17932/commits/cf35a018c18c33f288c251f0a8d7a0d16ae0814d.

In https://github.com/directus/directus/pull/17932/commits/7271508aebec68a8ebd2528fc4ffd4e30202ecb1 I wrapped the tests in a `try...finally` statement to ensure the listeners are also getting cleaned up if the tests fails (e.g. timeout reached). But I think we can ignore that in this context 👍

licitdev commented on tests-blackbox/common/test-logger.ts:
```diff
@@ -0,0 +1,66 @@
+	private stopped?: boolean;
+	private resolve?: (log: string) => void;
+
+	constructor(server: ChildProcess, stopCondition: string | StopFunction, filter?: string | FilterFunction);
+	constructor(server: ChildProcess, stopCondition: string, filter?: boolean | string | FilterFunction);
+	constructor(server: ChildProcess, stopCondition: string | StopFunction, filter?: boolean | string | FilterFunction) {
```
> So far all log entries of the server were captured and checked, which could theoretically lead to false positives, since the conditions in the `expect`s could also be met by other requests

Yes, having `filter` will prevent such theoretical error. However, I think that we should KISS.

Can we simplify `filter` to be just `boolean | string`? 
The usage of boolean to prevent repeated string is a good idea. 👍 
Same for the `stopCondition`, can we revert to `string`?

> I wrapped the tests in a `try...finally` statement to ensure the listeners are also getting cleaned up if the tests fails (e.g. timeout reached). 

Yes, it's no longer required since the extracted implementation in https://github.com/directus/directus/pull/17932/commits/03efdbc5ea0e0728ec59a7cf474bed474cb260f7

paescuj replied:
> Yes, having `filter` will prevent such theoretical error. However, I think that we should KISS.
> 
> Can we simplify `filter` to be just `boolean | string`? The usage of boolean to prevent repeated string is a good idea. 👍 Same for the `stopCondition`, can we revert to `string`?

Agree with KISS, adjusted now 👍 

> Yes, it's no longer required since the extracted implementation in https://github.com/directus/directus/commit/03efdbc5ea0e0728ec59a7cf474bed474cb260f7

What I meant is that it doesn't work anymore, because `cleanup` might not be called if a test fails for whatever reason and the listener remains open. But again I think we can safely disregard this, since it is only the test environment and not the productive application. Just wanted to mention it 😃

paescuj committed:
Simplify again, can still be extended later on

KISS FTW

paescuj committed:
Only ignore `uploads` in root
